.github/workflows/e2e.yml

```diff
@@ -1,9 +1,7 @@
 name: E2E
 
 on:
-  push:
```

Comment:
I don't remember exactly why it's set on push, but remember there used to be a good reason.

Reply:
We can add the CI to push which would include e2e and all the other checks too.

Reply:
I found the answer, it's E2E that will not work for PR events and only push, because of the matching branch functionality: https://github.com/opencollective/opencollective-api/pull/3186

Reply:
I update this PR to handle the matching branches case for both push and pull_request types